src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -94,7 +96,25 @@ struct primitive_impl {
     virtual void set_kernels(cldnn::kernels_cache::compiled_kernels kernels) {}
     virtual std::vector<kernel::ptr> get_kernels() { return {}; }
 
+    bool need_weights_reorder() const { return _weights_reorder_params != nullptr; }
+    std::shared_ptr<WeightsReorderParams> get_weights_reorder_params() const { return _weights_reorder_params; }
+    void reset_weights_reorder_params() { _weights_reorder_params = nullptr; }
+
+    std::shared_ptr<kernel_impl_params> get_weights_reorder_kernel_params() const {
```

Comment:
Could we move the code body to primitive_inst.cpp to reduce future build time?

Reply:
Sure, moved to primitive_inst.cpp